vanetza/security/straight_verify_service.cpp

```diff
@@ -431,6 +431,14 @@ VerifyConfirm StraightVerifyService::verify(const v3::SecuredMessage& msg)
         return confirm;
     }
 
+    ByteBuffer encoded_signing_payload;
+    try {
+        encoded_signing_payload = msg.signing_payload();
+    } catch (...) {
+        confirm.report = VerificationReport::False_Signature;
```

Comment:
Maybe we should add a more descriptive report value, e.g. `Corrupt_Message` or `Encoding_Failure`?

Reply:
`Corrupt_Message` sounds good, I can do that. I have seen that in `security::DelegatingSecurityEntity::decapsulate_packet:39`, `VerificationReport` from the verify services is cast to `DecapReport` from SN-Decaps. Is it intentional that `Unencrypted_Message` and `Decryption_Error` is not included in `VerificationReport`? `Decryption_Error` is maybe another good candidate.

Unfortunately, I haven't found anything about this, but is it perhaps possible with asn1c to extract the indices and length of certain elements? That way, we could skip the entire encode step.

Reply:
Good catch regarding the report casting. For the sake of robustness, this cast should be replaced by a proper mapping function.

I am not aware of any smart way to get the length of asn1c data structures without actually calling the encoder. You can call the encoding stage with a "null sink" though, i.e. no actual bytes are written only the "written bytes" are tracked.

Reply:
I have annotated the enums with values according to [AUTOSAR SWS_V2xM_91000](https://www.autosar.org/fileadmin/standards/R22-11/CP/AUTOSAR_SWS_V2XManagement.pdf) so that the cast at least remains consistent. I hope it's okay that I added the None enum according to the AUTOSAR specification.